From 6afd8bac7f8e7b95984b5a1be4bb5ec71c33f44f Mon Sep 17 00:00:00 2001
Message-ID: <6afd8bac7f8e7b95984b5a1be4bb5ec71c33f44f.1753477523.git.sam@gentoo.org>
From: Florian Weimer <fweimer@redhat.com>
Date: Fri, 2 May 2025 17:41:43 +0200
Subject: [PATCH 1/2] libsanitizer: Fix build with glibc 2.42

The termio structure will be removed from glibc 2.42.  It has
been deprecated since the late 80s/early 90s.

Cherry-picked from LLVM commit 59978b21ad9c65276ee8e14f26759691b8a65763
("[sanitizer_common] Remove interceptors for deprecated struct termio
(#137403)").

Co-Authored-By: Tom Stellard <tstellar@redhat.com>

libsanitizer/

	* sanitizer_common/sanitizer_common_interceptors_ioctl.inc: Cherry
	picked from LLVM commit 59978b21ad9c65276ee8e14f26759691b8a65763.
	* sanitizer_common/sanitizer_platform_limits_posix.cpp: Likewise.
	* sanitizer_common/sanitizer_platform_limits_posix.h: Likewise.

(cherry picked from commit 1789c57dc97ea2f9819ef89e28bf17208b6208e7)
---
 .../sanitizer_common_interceptors_ioctl.inc               | 8 --------
 .../sanitizer_common/sanitizer_platform_limits_posix.cpp  | 3 ---
 .../sanitizer_common/sanitizer_platform_limits_posix.h    | 1 -
 3 files changed, 12 deletions(-)

Adapted from gcc12 to gcc-4.9.4 by Chris Severance aur.severach aATt spamgourmet dott com

diff -pNaru3 a/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc b/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc
--- a/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc	2013-12-05 04:18:38.000000000 -0500
+++ b/libsanitizer/sanitizer_common/sanitizer_common_interceptors_ioctl.inc	2025-10-24 23:01:56.699644218 -0400
@@ -318,13 +318,9 @@ static void ioctl_table_fill() {
   _(SOUND_PCM_WRITE_CHANNELS, WRITE, sizeof(int));
   _(SOUND_PCM_WRITE_FILTER, WRITE, sizeof(int));
   _(TCFLSH, NONE, 0);
-  _(TCGETA, WRITE, struct_termio_sz);
   _(TCGETS, WRITE, struct_termios_sz);
   _(TCSBRK, NONE, 0);
   _(TCSBRKP, NONE, 0);
-  _(TCSETA, READ, struct_termio_sz);
-  _(TCSETAF, READ, struct_termio_sz);
-  _(TCSETAW, READ, struct_termio_sz);
   _(TCSETS, READ, struct_termios_sz);
   _(TCSETSF, READ, struct_termios_sz);
   _(TCSETSW, READ, struct_termios_sz);
diff -pNaru3 a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc
--- a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc	2025-10-24 23:01:23.106001186 -0400
+++ b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.cc	2025-10-24 23:01:56.699644218 -0400
@@ -298,7 +298,6 @@ namespace __sanitizer {
   unsigned struct_sbi_instrument_sz = sizeof(struct sbi_instrument);
   unsigned struct_seq_event_rec_sz = sizeof(struct seq_event_rec);
   unsigned struct_synth_info_sz = sizeof(struct synth_info);
-  unsigned struct_termio_sz = sizeof(struct termio);
   unsigned struct_vt_consize_sz = sizeof(struct vt_consize);
   unsigned struct_vt_mode_sz = sizeof(struct vt_mode);
   unsigned struct_vt_sizes_sz = sizeof(struct vt_sizes);
diff -pNaru3 a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h
--- a/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h	2025-10-24 23:01:23.106001186 -0400
+++ b/libsanitizer/sanitizer_common/sanitizer_platform_limits_posix.h	2025-10-24 23:01:56.699644218 -0400
@@ -598,7 +598,6 @@ namespace __sanitizer {
   extern unsigned struct_sbi_instrument_sz;
   extern unsigned struct_seq_event_rec_sz;
   extern unsigned struct_synth_info_sz;
-  extern unsigned struct_termio_sz;
   extern unsigned struct_vt_consize_sz;
   extern unsigned struct_vt_mode_sz;
   extern unsigned struct_vt_sizes_sz;
